vaadin-chip-combobox/src/main/java/software/xdev/vaadin/chips/ChipComboBox.java

```diff
@@ -646,4 +654,56 @@ public FlexLayout getChipsContainer()
 	{
 		return this.chipsContainer;
 	}
+	
+	/**
+	 * Simply attaches a listener to the dataprovider if data changes and sets the provided items through
+	 * {@link #setItems(Collection)}
+	 *
+	 * @param dataProvider DataProvider instance to use, not <code>null</code>
+	 * @return the generic {@link DataView} implementation for ComboBox
+	 */
+	@Override
+	public ComboBoxDataView<T> setItems(final DataProvider<T, String> dataProvider)
+	{
+		dataProvider.addDataProviderListener(this::dataProviderUpdated);
+		this.setItems(this.extractItemsFromDataProvider(dataProvider));
+		return this.cbAvailableItems.getGenericDataView();
+	}
+	
+	/**
+	 * Simply attaches a listener to the dataprovider if data changes and sets the provided items through
+	 * {@link #setItems(Collection)}
+	 *
+	 * @param dataProvider DataProvider instance to use, not <code>null</code>
+	 * @return the generic {@link DataView} implementation for ComboBox
+	 */
+	@Override
+	public ComboBoxDataView<T> setItems(
+		final InMemoryDataProvider<T> dataProvider)
+	{
+		dataProvider.addDataProviderListener(this::dataProviderUpdated);
+		this.setItems(this.extractItemsFromDataProvider(dataProvider));
+		return this.cbAvailableItems.getGenericDataView();
+	}
+	
+	private void dataProviderUpdated(final DataChangeEvent<T> tDataChangeEvent)
+	{
+		this.setItems(this.extractItemsFromDataProvider(tDataChangeEvent.getSource()));
+	}
+	
+	private List<T> extractItemsFromDataProvider(final DataProvider<T, ?> dataProvider)
+	{
+		return dataProvider.fetch(new Query<>()).collect(Collectors.toList());
+	}
```

Comment:
I don't think we should do this like this. Here we're reading all the data from the Dataprovider.

A dataprovider can contain huge amounts of data and was likely designed with the thought that not all that data should be in memory at once.

Is there a different solution for handling the problem? Because this solution yields no improvements in my eyes.